core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1548,6 +1548,7 @@ public void delete(AlluxioURI path, DeleteContext context)
             for (LockedInodePath childPath : descendants) {
               try {
                 mPermissionChecker.checkPermission(Mode.Bits.WRITE, childPath);
+                mMountTable.checkUnderWritableMountPoint(childPath.getUri());
```

Comment:
How is this different from the check below? https://github.com/Alluxio/alluxio/pull/11513/files#diff-7f82122f923de058e3c4a20b43cd7e4eR1565

Reply:
The check below won't check descendants of the path, it just check the path itself

Reply:
I see. Would checking only the descendant mount points work? I am concerned that checking every path may be unnecessarily expensive, especially for a large sub-tree of the namespace.

Reply:
We already check the permissions of every file and directory when deleting recursively. I don't think this would make much of a difference

Reply:
I thought it was unnecessary to check the same mount point multiple times. Also, `checkUnderWritableMountPoint()` calls `getMountPoint()`, and that needs to iterate over all the mount points to find the longest prefix.

Reply:
> We already check the permissions of every file and directory when deleting recursively. I don't think this would make much of a difference

Well, I've updated the use-case in my first comment. As it described, the scenario is, a user with the right permission did this operation, the should-be-protected sub-directory was delete unexpectedly

Reply:
> I thought it was unnecessary to check the same mount point multiple times. Also, `checkUnderWritableMountPoint()` calls `getMountPoint()`, and that needs to iterate over all the mount points to find the longest prefix.

Thanks for your advice! I think checking only the descendant mount points works